src/extendStyles.jsx

```diff
@@ -0,0 +1,46 @@
+/* eslint react/forbid-foreign-prop-types: off */
+
+import deepmerge from 'deepmerge';
+
+// Recursively iterate through the style object, validating the same path exists in the
+// extendableStyles object.
+function validateStyle(style, extendableStyles, path = '') {
+  if (process.env.NODE_ENV !== 'production') {
+    // Stop recursively validating when we hit a style's value
+    if (typeof style === 'string' || typeof style === 'number') {
+      return;
+    }
+
+    const styleKeys = Object.keys(style);
+    if (styleKeys.length) {
+      styleKeys.forEach((styleKey) => {
+        const currentPath = `${path}.${styleKey}`;
+        const isValid = extendableStyles[styleKey];
+        if (!isValid) {
+          throw new Error(
+            `withStyles() extending style is invalid: ${currentPath}. If this style is expected, add it to`
+            + 'the component\'s "extendableStyles" option.',
```

Comment:
this can be collapsed onto the previous line; max-len ignores long lines with strings.